src/fiatconnect/index.ts

```diff
@@ -29,6 +55,67 @@ export async function getFiatConnectProviders(): Promise<FiatConnectClientConfig
   return providers
 }
 
+type QuotesInput = {
+  fiatConnectProviders: FiatConnectProviderInfo[]
+  flow: CICOFlow
+  localCurrency: LocalCurrencyCode
+  digitalAsset: CiCoCurrency
+  cryptoAmount: number
+  country: string
+}
+
+type GetFiatConnectQuotesResponse = {
+  id: string
+  ok: boolean
+  val: QuoteResponse | QuoteErrorResponse | { error: string }
+}
+
+export type FiatConnectQuoteError = {
+  provider: FiatConnectProviderInfo
+  ok: boolean
+} & (QuoteErrorResponse | { error: string })
+
+export type FiatConnectQuoteSuccess = {
+  provider: FiatConnectProviderInfo
+  ok: boolean
+} & QuoteResponse
+
+export async function getFiatConnectQuotes(
+  params: QuotesInput
+): Promise<(FiatConnectQuoteSuccess | FiatConnectQuoteError)[]> {
+  const { fiatConnectProviders, localCurrency, digitalAsset, cryptoAmount, country, flow } = params
+  const fiatType = convertToFiatConnectFiatCurrency(localCurrency)
+  const cryptoType = convertToFiatConnectCryptoCurrency(digitalAsset)
+  if (!fiatType || !cryptoType) return []
+  const quoteParams: QuoteRequestQuery = {
+    fiatType,
+    cryptoType,
+    cryptoAmount: cryptoAmount.toString(),
+    country,
+  }
+  const providers = fiatConnectProviders.map((provider) => provider.id).join(',')
+  const queryParams = new URLSearchParams({
+    ...(quoteParams as Record<string, string>),
+    providers,
+    quoteType: flow === CICOFlow.CashIn ? 'in' : 'out',
+  }).toString()
+  const response = await fetch(`${networkConfig.getFiatConnectQuotesUrl}?${queryParams}`)
+  if (!response.ok) {
+    const err = await response.json()
+    Logger.error(TAG, `Failure response fetching FiatConnect quotes: ${err} , returning empty list`)
+    return []
+  }
+  const results: { quotes: GetFiatConnectQuotesResponse[] } = await response.json()
+  return results.quotes
+    .filter((quote) => quote.ok)
+    .map((result) => ({
+      ...result.val,
+      ok: result.ok,
+      provider: fiatConnectProviders.find(
+        (provider) => provider.id === result.id
+      ) as FiatConnectProviderInfo,
```

Comment:
You could shorthand this as 
```suggestion
      provider: fiatConnectProviders.find((provider) => provider.id === result.id)!,
```